base/compiler/stmtinfo.jl

```diff
@@ -48,20 +51,27 @@ each partition (`info.matches::Vector{MethodMatchInfo}`).
 This info is illegal on any statement that is not a call to a generic function.
 """
 struct UnionSplitInfo <: CallInfo
-    matches::Vector{MethodMatchInfo}
+    matches::Vector{MethodLookupResult}
+    mts::Vector{MethodTable}
+    fullmatches::Vector{Bool}
```

Comment:
Would it be okay to revert back to storing `Vector{MethodMatchInfo}` like before? The current design, where `matches`/`mts`/`fullmatches` can have different lengths, feels a bit tricky to handle. I understand it's meant to avoid adding duplicate edges to the same `mt` in `add_uncovered_edges_impl`, but in practice, adding duplicate edges isn't really an issue. From a complexity standpoint, I think the previous approach is preferable.

Reply:
How would you feel about:
```suggestion
    matches::Vector{MethodLookupResult}
    mt_edges::Vector{@NamedTuple{mt::MethodTable, fullmatch::Bool}}
```

or similar?

That's more consistent with what `UnionSplitMethodMatches` carried before, which always had these edges de-duplicated.

Reply:
Right, the previous design of `UnionSplitMethodMatches` also avoided adding duplicated edges. But I still feel that it's better for `UnionSplitInfo` or `UnionSplitMethodMatches` to have a simpler data structure. We can avoid duplication when adding edges.

Reply:
Sure, feel free to open a PR